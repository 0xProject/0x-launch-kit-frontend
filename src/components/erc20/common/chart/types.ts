export interface Cord {
    x: number;
    y: number;
}
export enum OrderSide {
    Bid,
    Ask,
}

export interface Margin {
    left: number;
    right: number;
    top: number;
    bottom: number;
}

export interface ChartDimensions {
    height: number;
    margin: Margin;
}
