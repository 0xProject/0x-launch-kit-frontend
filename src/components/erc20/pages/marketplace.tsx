import React from 'react';

import { CheckMetamaskStateModalContainer } from '../../common/check_metamask_state_modal_container';
import { ColumnNarrow } from '../../common/column_narrow';
import { ColumnWide } from '../../common/column_wide';
import { Content } from '../common/content_wrapper';
import { BuySellContainer } from '../marketplace/buy_sell';
import { OrderBookTableContainer } from '../marketplace/order_book';
import { OrderHistoryContainer } from '../marketplace/order_history';
import { WalletBalanceContainer } from '../marketplace/wallet_balance';

class Marketplace extends React.PureComponent {
    public render = () => {
        return (
            <Content>
                <ColumnNarrow>
                    <WalletBalanceContainer />
                    <BuySellContainer />
                </ColumnNarrow>
                <ColumnNarrow>
                    <OrderBookTableContainer />
                </ColumnNarrow>
                <ColumnWide>
                    <OrderHistoryContainer />
                </ColumnWide>
                <CheckMetamaskStateModalContainer />
            </Content>
        );
    };
}

export { Marketplace };
