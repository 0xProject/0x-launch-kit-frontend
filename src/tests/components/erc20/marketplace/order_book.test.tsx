/**
 * @jest-environment jsdom
 */

import { BigNumber } from '0x.js';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';

import { OrderBookTableWithTheme } from '../../../../components/erc20/marketplace/order_book';
import { DefaultTheme } from '../../../../themes/default_theme';
import { openOrder, tokenFactory } from '../../../../util/test-utils';
import { OrderSide, Token, Web3State } from '../../../../util/types';

const mockStore = configureMockStore([]);

const theme = new DefaultTheme();

describe('OrderBookTable', () => {
    const absoluteSpread = new BigNumber('0.03');
    const percentageSpread = new BigNumber('3');

    it('Renders my size column with value', () => {
        // given
        const orderBook = {
            sellOrders: [
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.02'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('4'),
                    price: new BigNumber('1.01'),
                },
            ],
            buyOrders: [
                {
                    side: OrderSide.Buy,
                    size: new BigNumber('2'),
                    price: new BigNumber('0.5'),
                },
            ],
            mySizeOrders: [
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
            ],
        };

        const baseToken = tokenFactory.build();
        const quoteToken = tokenFactory.build();

        const userOrder1 = openOrder({
            side: OrderSide.Buy,
            size: new BigNumber(2),
            price: new BigNumber('0.5'),
        });

        const userOrders = [userOrder1];

        const store = mockStore({});

        // when
        const wrapper = (
            <ThemeProvider theme={theme}>
                <Provider store={store}>
                    <OrderBookTableWithTheme
                        orderBook={orderBook}
                        baseToken={baseToken}
                        quoteToken={quoteToken}
                        userOrders={userOrders}
                        absoluteSpread={absoluteSpread}
                        percentageSpread={percentageSpread}
                    />
                </Provider>
            </ThemeProvider>
        );

        const tree = renderer.create(wrapper).toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('Should render a row of mySize with the total amount for one order', () => {
        // Given
        const orderBook = {
            sellOrders: [
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('4'),
                    price: new BigNumber('1.01'),
                },
            ],
            buyOrders: [
                {
                    side: OrderSide.Buy,
                    size: new BigNumber('2'),
                    price: new BigNumber('0.5'),
                },
            ],
            mySizeOrders: [
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
            ],
        };

        const baseToken = tokenFactory.build();
        const quoteToken = tokenFactory.build();

        const userOrder1 = openOrder({
            side: OrderSide.Buy,
            size: new BigNumber(2),
            price: new BigNumber('0.5'),
        });

        const userOrders = [userOrder1];

        const store = mockStore({});

        // when
        const wrapper = (
            <ThemeProvider theme={theme}>
                <Provider store={store}>
                    <OrderBookTableWithTheme
                        orderBook={orderBook}
                        baseToken={baseToken}
                        quoteToken={quoteToken}
                        userOrders={userOrders}
                        percentageSpread={percentageSpread}
                        absoluteSpread={absoluteSpread}
                    />
                </Provider>
            </ThemeProvider>
        );

        const tree = renderer.create(wrapper).toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('Check if my size renders an item with more than two decimals', () => {
        // Given
        const orderBook = {
            sellOrders: [
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1.1234567'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1.123456'),
                    price: new BigNumber('1.02'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1.12345'),
                    price: new BigNumber('1.01'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1.1234'),
                    price: new BigNumber('1.01'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1.123'),
                    price: new BigNumber('1.01'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1.12'),
                    price: new BigNumber('1.01'),
                },
            ],
            buyOrders: [
                {
                    side: OrderSide.Buy,
                    size: new BigNumber('2'),
                    price: new BigNumber('0.5'),
                },
            ],
            mySizeOrders: [
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
            ],
        };

        const baseToken = tokenFactory.build();
        const quoteToken = tokenFactory.build();

        const userOrder1 = openOrder({
            side: OrderSide.Buy,
            size: new BigNumber(2),
            price: new BigNumber('0.5'),
        });

        const userOrders = [userOrder1];

        const store = mockStore({});

        // when
        const wrapper = (
            <ThemeProvider theme={theme}>
                <Provider store={store}>
                    <OrderBookTableWithTheme
                        orderBook={orderBook}
                        baseToken={baseToken}
                        quoteToken={quoteToken}
                        userOrders={userOrders}
                        absoluteSpread={absoluteSpread}
                        percentageSpread={percentageSpread}
                    />
                </Provider>
            </ThemeProvider>
        );

        // then
        const tree = renderer.create(wrapper).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('Should not render my size column if the user does not have metamask', () => {
        const orderBook = {
            sellOrders: [
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('4'),
                    price: new BigNumber('1.01'),
                },
            ],
            buyOrders: [
                {
                    side: OrderSide.Buy,
                    size: new BigNumber('2'),
                    price: new BigNumber('0.5'),
                },
            ],
            mySizeOrders: [
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
            ],
        };

        const token: Token = {
            address: '0x871dd7c2b4b25e1aa18728e9d5f2af4c4e431f5c',
            decimals: 0,
            name: '0x',
            symbol: 'zrx',
            primaryColor: '#ccc',
            displayDecimals: 2,
        };

        const userOrder1 = openOrder({
            side: OrderSide.Buy,
            size: new BigNumber(2),
            price: new BigNumber('0.5'),
        });

        const userOrders = [userOrder1];

        const store = mockStore({});

        // when
        const wrapper = (
            <ThemeProvider theme={theme}>
                <Provider store={store}>
                    <OrderBookTableWithTheme
                        orderBook={orderBook}
                        baseToken={token}
                        quoteToken={token}
                        userOrders={userOrders}
                        web3State={Web3State.NotInstalled}
                        absoluteSpread={absoluteSpread}
                        percentageSpread={percentageSpread}
                    />
                </Provider>
            </ThemeProvider>
        );

        // then
        const tree = renderer.create(wrapper).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('Should not render my size column if the user rejected the metamask permissions', () => {
        const orderBook = {
            sellOrders: [
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('4'),
                    price: new BigNumber('1.01'),
                },
            ],
            buyOrders: [
                {
                    side: OrderSide.Buy,
                    size: new BigNumber('2'),
                    price: new BigNumber('0.5'),
                },
            ],
            mySizeOrders: [
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
                {
                    side: OrderSide.Sell,
                    size: new BigNumber('1'),
                    price: new BigNumber('1.5'),
                },
            ],
        };

        const token: Token = {
            address: '0x871dd7c2b4b25e1aa18728e9d5f2af4c4e431f5c',
            decimals: 0,
            name: '0x',
            symbol: 'zrx',
            primaryColor: '#ccc',
            displayDecimals: 2,
        };

        const userOrder1 = openOrder({
            side: OrderSide.Buy,
            size: new BigNumber(2),
            price: new BigNumber('0.5'),
        });

        const userOrders = [userOrder1];

        const store = mockStore({});

        // when
        const wrapper = (
            <ThemeProvider theme={theme}>
                <Provider store={store}>
                    <OrderBookTableWithTheme
                        orderBook={orderBook}
                        baseToken={token}
                        quoteToken={token}
                        userOrders={userOrders}
                        web3State={Web3State.Locked}
                        absoluteSpread={absoluteSpread}
                        percentageSpread={percentageSpread}
                    />
                </Provider>
            </ThemeProvider>
        );

        // then
        const tree = renderer.create(wrapper).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
