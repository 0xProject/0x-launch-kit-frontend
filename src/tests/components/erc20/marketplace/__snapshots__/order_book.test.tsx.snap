// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBookTable Check if my size renders an item with more than two decimals 1`] = `
<div
  className="sc-kpOJdX kUlmMq sc-bwzfXH cfOWoU sc-bdVaJa fJeBhq"
>
  <div
    className="sc-htpNat fMPtTX"
  >
    <h1
      className="sc-bxivhb DvYdZ"
    >
      Orderbook
    </h1>
  </div>
  <div
    className="sc-ifAKCX jlvJWy"
  >
    <div
      className="sc-dxgOiQ sc-jKJlTe kauRMH"
    >
      <div
        className="sc-VigVT jFeuG"
      >
        Trade size
      </div>
      <div
        className="sc-VigVT jFeuG"
      >
        Price (
        MOCK6
        )
      </div>
      <div
        className="sc-VigVT sc-chPdSV ePzcBE"
      >
        My Size
      </div>
    </div>
    <div
      className="sc-hMqMXs iwcMlS"
      onScroll={[Function]}
    >
      <div
        className="sc-kEYyzF fKDJln"
      >
        <div
          className="sc-kkGfuU sc-iAyFgw daOdFJ"
        >
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-fBuWsC kwGDhd"
              >
                1.1235
              </span>
              <span
                className="sc-fMiknA fWien"
              >
                
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.5000000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG dwncel"
              id="mySize"
            >
              1.0000
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-dVhcbM kICZVG"
              >
                1.1235
              </span>
              <span
                className="sc-eqIVtm LdBiZ"
              >
                
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.0200000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-fAjcbJ hRmYGJ"
              >
                1.1235
              </span>
              <span
                className="sc-caSCKo rrezN"
              >
                
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.0100000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-gisBJw bPPCsh"
              >
                1.1234
              </span>
              <span
                className="sc-kjoXOD coUgfy"
              >
                
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.0100000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-cHGsZl iyCOMG"
              >
                1.123
              </span>
              <span
                className="sc-TOsTZ kBTbna"
              >
                0
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.0100000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-kgAjT kXvHqg"
              >
                1.12
              </span>
              <span
                className="sc-cJSrbW imkuMm"
              >
                00
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.0100000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
        </div>
        <div
          className="sc-kgoBCf sc-kGXeez jmVvb"
        >
          <div
            className="sc-jTzLTM sc-cSHVUG kdXjDv"
          >
            Spread
          </div>
          <div
            className="sc-jTzLTM kcTCzl"
          >
            0.0300000
          </div>
          <div
            className="sc-jTzLTM sc-jzJRlG fnqkDA"
          >
            3.00
            %
          </div>
        </div>
        <div
          className="sc-kkGfuU sc-hSdWYo dhGrlp"
        >
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-ksYbfQ fVlPYy"
              >
                2
              </span>
              <span
                className="sc-hmzhuo hAdYWW"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM eTqOuf"
            >
              0.5000000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OrderBookTable Renders my size column with value 1`] = `
<div
  className="sc-kpOJdX kUlmMq sc-bwzfXH cfOWoU sc-bdVaJa fJeBhq"
>
  <div
    className="sc-htpNat fMPtTX"
  >
    <h1
      className="sc-bxivhb DvYdZ"
    >
      Orderbook
    </h1>
  </div>
  <div
    className="sc-ifAKCX jlvJWy"
  >
    <div
      className="sc-dxgOiQ sc-jKJlTe kauRMH"
    >
      <div
        className="sc-VigVT jFeuG"
      >
        Trade size
      </div>
      <div
        className="sc-VigVT jFeuG"
      >
        Price (
        MOCK2
        )
      </div>
      <div
        className="sc-VigVT sc-chPdSV ePzcBE"
      >
        My Size
      </div>
    </div>
    <div
      className="sc-hMqMXs iwcMlS"
      onScroll={[Function]}
    >
      <div
        className="sc-kEYyzF fKDJln"
      >
        <div
          className="sc-kkGfuU sc-iAyFgw daOdFJ"
        >
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-eHgmQL dPxpcD"
              >
                1
              </span>
              <span
                className="sc-cvbbAY bvjWlU"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.5000000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG dwncel"
              id="mySize"
            >
              1.0000
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-jWBwVP iUEkdW"
              >
                1
              </span>
              <span
                className="sc-brqgnP hIEAPA"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.0200000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-cMljjf fKSpyD"
              >
                4
              </span>
              <span
                className="sc-jAaTju fAkqWE"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.0100000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
        </div>
        <div
          className="sc-kgoBCf sc-kGXeez jmVvb"
        >
          <div
            className="sc-jTzLTM sc-cSHVUG kdXjDv"
          >
            Spread
          </div>
          <div
            className="sc-jTzLTM kcTCzl"
          >
            0.0300000
          </div>
          <div
            className="sc-jTzLTM sc-jzJRlG fnqkDA"
          >
            3.00
            %
          </div>
        </div>
        <div
          className="sc-kkGfuU sc-hSdWYo dhGrlp"
        >
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-jDwBTQ fvUbIf"
              >
                2
              </span>
              <span
                className="sc-gPEVay hknufd"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM eTqOuf"
            >
              0.5000000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OrderBookTable Should not render my size column if the user does not have metamask 1`] = `
<div
  className="sc-kpOJdX kUlmMq sc-bwzfXH cfOWoU sc-bdVaJa fJeBhq"
>
  <div
    className="sc-htpNat fMPtTX"
  >
    <h1
      className="sc-bxivhb DvYdZ"
    >
      Orderbook
    </h1>
  </div>
  <div
    className="sc-ifAKCX jlvJWy"
  >
    <div
      className="sc-dxgOiQ sc-jKJlTe kauRMH"
    >
      <div
        className="sc-VigVT jFeuG"
      >
        Trade size
      </div>
      <div
        className="sc-VigVT jFeuG"
      >
        Price (
        zrx
        )
      </div>
    </div>
    <div
      className="sc-hMqMXs iwcMlS"
      onScroll={[Function]}
    >
      <div
        className="sc-kEYyzF fKDJln"
      >
        <div
          className="sc-kkGfuU sc-iAyFgw daOdFJ"
        >
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-frDJqD dwYjGo"
              >
                1
              </span>
              <span
                className="sc-kvZOFW gSNEEv"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.5000000
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-hqyNC kMXIrR"
              >
                1
              </span>
              <span
                className="sc-jbKcbu iQDXnV"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.5000000
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-dNLxif dMzXPz"
              >
                4
              </span>
              <span
                className="sc-jqCOkK fYUDXS"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.0100000
            </div>
          </div>
        </div>
        <div
          className="sc-kgoBCf sc-kGXeez jmVvb"
        >
          <div
            className="sc-jTzLTM sc-cSHVUG kdXjDv"
          >
            Spread
          </div>
          <div
            className="sc-jTzLTM kcTCzl"
          >
            0.0300000
          </div>
          <div
            className="sc-jTzLTM sc-jzJRlG fnqkDA"
          >
            3.00
            %
          </div>
        </div>
        <div
          className="sc-kkGfuU sc-hSdWYo dhGrlp"
        >
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-uJMKN gGvEOz"
              >
                2
              </span>
              <span
                className="sc-bbmXgH kPOpfp"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM eTqOuf"
            >
              0.5000000
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OrderBookTable Should not render my size column if the user rejected the metamask permissions 1`] = `
<div
  className="sc-kpOJdX kUlmMq sc-bwzfXH cfOWoU sc-bdVaJa fJeBhq"
>
  <div
    className="sc-htpNat fMPtTX"
  >
    <h1
      className="sc-bxivhb DvYdZ"
    >
      Orderbook
    </h1>
  </div>
  <div
    className="sc-ifAKCX jlvJWy"
  >
    <div
      className="sc-dxgOiQ sc-jKJlTe kauRMH"
    >
      <div
        className="sc-VigVT jFeuG"
      >
        Trade size
      </div>
      <div
        className="sc-VigVT jFeuG"
      >
        Price (
        zrx
        )
      </div>
    </div>
    <div
      className="sc-hMqMXs iwcMlS"
      onScroll={[Function]}
    >
      <div
        className="sc-kEYyzF fKDJln"
      >
        <div
          className="sc-kkGfuU sc-iAyFgw daOdFJ"
        >
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-gGBfsJ gjvvyH"
              >
                1
              </span>
              <span
                className="sc-jnlKLf hUZhox"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.5000000
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-fYxtnH igTHrv"
              >
                1
              </span>
              <span
                className="sc-tilXH gaRzoJ"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.5000000
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-hEsumM flURld"
              >
                4
              </span>
              <span
                className="sc-ktHwxA fBHFKN"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.0100000
            </div>
          </div>
        </div>
        <div
          className="sc-kgoBCf sc-kGXeez jmVvb"
        >
          <div
            className="sc-jTzLTM sc-cSHVUG kdXjDv"
          >
            Spread
          </div>
          <div
            className="sc-jTzLTM kcTCzl"
          >
            0.0300000
          </div>
          <div
            className="sc-jTzLTM sc-jzJRlG fnqkDA"
          >
            3.00
            %
          </div>
        </div>
        <div
          className="sc-kkGfuU sc-hSdWYo dhGrlp"
        >
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-cIShpX kjZQlS"
              >
                2
              </span>
              <span
                className="sc-kafWEX dmXPxV"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM eTqOuf"
            >
              0.5000000
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`OrderBookTable Should render a row of mySize with the total amount for one order 1`] = `
<div
  className="sc-kpOJdX kUlmMq sc-bwzfXH cfOWoU sc-bdVaJa fJeBhq"
>
  <div
    className="sc-htpNat fMPtTX"
  >
    <h1
      className="sc-bxivhb DvYdZ"
    >
      Orderbook
    </h1>
  </div>
  <div
    className="sc-ifAKCX jlvJWy"
  >
    <div
      className="sc-dxgOiQ sc-jKJlTe kauRMH"
    >
      <div
        className="sc-VigVT jFeuG"
      >
        Trade size
      </div>
      <div
        className="sc-VigVT jFeuG"
      >
        Price (
        MOCK4
        )
      </div>
      <div
        className="sc-VigVT sc-chPdSV ePzcBE"
      >
        My Size
      </div>
    </div>
    <div
      className="sc-hMqMXs iwcMlS"
      onScroll={[Function]}
    >
      <div
        className="sc-kEYyzF fKDJln"
      >
        <div
          className="sc-kkGfuU sc-iAyFgw daOdFJ"
        >
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-iRbamj jAFnxN"
              >
                1
              </span>
              <span
                className="sc-jlyJG djZUoc"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.5000000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG dwncel"
              id="mySize"
            >
              2.0000
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-gipzik jsUTZA"
              >
                1
              </span>
              <span
                className="sc-csuQGl fVQmcy"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.5000000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG dwncel"
              id="mySize"
            >
              2.0000
            </div>
          </div>
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-Rmtcm dTnZHf"
              >
                4
              </span>
              <span
                className="sc-bRBYWo bCgeXn"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM kLxUPS"
            >
              1.0100000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
        </div>
        <div
          className="sc-kgoBCf sc-kGXeez jmVvb"
        >
          <div
            className="sc-jTzLTM sc-cSHVUG kdXjDv"
          >
            Spread
          </div>
          <div
            className="sc-jTzLTM kcTCzl"
          >
            0.0300000
          </div>
          <div
            className="sc-jTzLTM sc-jzJRlG fnqkDA"
          >
            3.00
            %
          </div>
        </div>
        <div
          className="sc-kkGfuU sc-hSdWYo dhGrlp"
        >
          <div
            className="sc-dxgOiQ sc-ckVGcZ fdRenD"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="sc-jTzLTM gYeEqk"
            >
              <span
                className="sc-hzDkRC gWNJWs"
              >
                2
              </span>
              <span
                className="sc-jhAzac eZYpSN"
              >
                .0000
              </span>
            </div>
            <div
              className="sc-jTzLTM eTqOuf"
            >
              0.5000000
            </div>
            <div
              className="sc-jTzLTM sc-jzJRlG fYJJdT"
              id="mySize"
            >
              -
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
